File: plugin-core/src/main/java/com/github/catatafishen/ideagentforcopilot/settings/BillingConfigurable.java
package com.github.catatafishen.ideagentforcopilot.settings;

import com.github.catatafishen.ideagentforcopilot.ui.CopilotBillingClient;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.options.Configurable;
import com.intellij.ui.HyperlinkLabel;
import com.intellij.ui.components.JBCheckBox;
import com.intellij.ui.components.JBLabel;
import com.intellij.ui.components.JBTextField;
import com.intellij.util.ui.FormBuilder;
import com.intellij.util.ui.JBUI;
import com.intellij.util.ui.UIUtil;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public final class BillingConfigurable implements Configurable {

    public static final String ID = "com.github.catatafishen.ideagentforcopilot.billing";

    private JBCheckBox showCopilotUsageCb;
    private JBTextField ghBinaryPathField;
    private BillingSettings settings;
    private JBLabel statusLabel;
    private JPanel mainPanel;

    @Override
    public @Nls(capitalization = Nls.Capitalization.Title) String getDisplayName() {
        return "Billing Data";
    }

    @Override
    public @NotNull JComponent createComponent() {
        settings = BillingSettings.getInstance();

        statusLabel = new JBLabel();
        ghBinaryPathField = new JBTextField();
        ghBinaryPathField.getEmptyText().setText("Auto-detect (leave empty)");

        JBLabel explanationLabel = new JBLabel(
            "<html><b>Why GitHub CLI is needed:</b><br/>" +
                "The Copilot ACP (Agent Communication Protocol) mode does not expose billing or usage data. " +
                "To view Copilot premium request usage, the plugin uses the GitHub CLI (<code>gh</code>) " +
                "to query GitHub's internal API endpoint.</html>");
        explanationLabel.setForeground(UIUtil.getContextHelpForeground());
        explanationLabel.setFont(JBUI.Fonts.smallFont());
        explanationLabel.setBorder(JBUI.Borders.emptyTop(4));

        HyperlinkLabel installLink = new HyperlinkLabel("Install GitHub CLI");
        installLink.setHyperlinkTarget("https://cli.github.com/");

        JBLabel installNote = new JBLabel(
            "<html>Install from <a href='https://cli.github.com/'>cli.github.com</a>, then authenticate with <code>gh auth login</code>.</html>");
        installNote.setForeground(UIUtil.getContextHelpForeground());
        installNote.setFont(JBUI.Fonts.smallFont());

        showCopilotUsageCb = new JBCheckBox("Show Copilot usage graph in toolbar");

        JButton recheckButton = new JButton("Recheck");
        recheckButton.addActionListener(e -> refreshGhCliStatusAsync());
        JPanel statusPanel = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 0));
        statusPanel.add(statusLabel);
        statusPanel.add(recheckButton);

        mainPanel = FormBuilder.createFormBuilder()
            .addComponent(new JBLabel(
                "<html>Configure how billing and usage data is displayed in the IDE.</html>"))
            .addSeparator(8)
            .addLabeledComponent("GitHub CLI Status:", statusPanel)
            .addLabeledComponent("GitHub CLI binary:", ghBinaryPathField)
            .addTooltip("Absolute path to gh CLI binary. Leave empty to auto-detect on PATH.")
            .addComponent(explanationLabel, 2)
            .addComponent(installNote, 2)
            .addComponent(installLink, 2)
            .addSeparator(12)
            .addComponent(showCopilotUsageCb)
            .addTooltip("Shows a usage graph icon in the main toolbar when Copilot usage data is available.")
            .addComponentFillVertically(new JPanel(), 0)
            .getPanel();
        mainPanel.setBorder(JBUI.Borders.empty(8));

        reset();
        return mainPanel;
    }

    @Override
    public boolean isModified() {
        if (showCopilotUsageCb.isSelected() != settings.isShowCopilotUsage()) return true;
        String currentGhPath = settings.getGhBinaryPath();
        String fieldText = ghBinaryPathField.getText().trim();
        return !fieldText.equals(currentGhPath == null ? "" : currentGhPath);
    }

    @Override
    public void apply() {
        settings.setShowCopilotUsage(showCopilotUsageCb.isSelected());
        String ghPath = ghBinaryPathField.getText().trim();
        settings.setGhBinaryPath(ghPath.isEmpty() ? null : ghPath);
    }

    @Override
    public void reset() {
        showCopilotUsageCb.setSelected(settings.isShowCopilotUsage());
        String ghPath = settings.getGhBinaryPath();
        ghBinaryPathField.setText(ghPath != null ? ghPath : "");
        refreshGhCliStatusAsync();
    }

    @Override
    public void disposeUIResources() {
        mainPanel = null;
        showCopilotUsageCb = null;
        ghBinaryPathField = null;
        statusLabel = null;
    }

    private void refreshGhCliStatusAsync() {
        if (statusLabel == null) return;
        statusLabel.setText("Checking...");
        statusLabel.setForeground(UIUtil.getLabelForeground());

        ApplicationManager.getApplication().executeOnPooledThread(() -> {
            CopilotBillingClient client = new CopilotBillingClient();
            String ghCli = client.findGhCli();
            boolean authenticated = ghCli != null && client.isGhAuthenticated(ghCli);

            SwingUtilities.invokeLater(() -> {
                if (statusLabel == null) return;
                if (ghCli == null) {
                    statusLabel.setText("GitHub CLI not found — install from cli.github.com");
                    statusLabel.setForeground(Color.RED);
                } else if (!authenticated) {
                    statusLabel.setText("GitHub CLI found but not authenticated — run 'gh auth login'");
                    statusLabel.setForeground(new Color(255, 140, 0)); // Orange
                } else {
                    statusLabel.setText("✓ GitHub CLI authenticated");
                    statusLabel.setForeground(new Color(0, 128, 0)); // Green
                }
            });
        });
    }
}

File: plugin-core/src/main/java/com/github/catatafishen/agentbridge/settings/BillingConfigurable.kt
package com.github.catatafishen.agentbridge.settings

import com.github.catatafishen.agentbridge.ui.CopilotBillingClient
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.options.BoundConfigurable
import com.intellij.openapi.options.SearchableConfigurable
import com.intellij.ui.HyperlinkLabel
import com.intellij.ui.JBColor
import com.intellij.ui.components.JBLabel
import com.intellij.ui.dsl.builder.AlignX
import com.intellij.ui.dsl.builder.MAX_LINE_LENGTH_WORD_WRAP
import com.intellij.ui.dsl.builder.bindSelected
import com.intellij.ui.dsl.builder.bindText
import com.intellij.ui.dsl.builder.panel
import com.intellij.util.ui.UIUtil

class BillingConfigurable :
    BoundConfigurable("Billing Data"),
    SearchableConfigurable {

    private val settings = BillingSettings.getInstance()
    private val statusLabel = JBLabel()

    override fun getId(): String = ID

    override fun createPanel() = panel {
        row {
            comment("Configure how billing and usage data is displayed in the IDE.")
        }
        separator()
        row("GitHub CLI Status:") {
            cell(statusLabel)
            button("Recheck") { refreshGhCliStatusAsync() }
        }
        row("GitHub CLI binary:") {
            textField()
                .align(AlignX.FILL)
                .resizableColumn()
                .applyToComponent { emptyText.text = "Auto-detect (leave empty)" }
                .comment("Absolute path to gh CLI binary. Leave empty to auto-detect on PATH.")
                .bindText(
                    { settings.ghBinaryPath.orEmpty() },
                    { settings.ghBinaryPath = it.trim().ifEmpty { null } }
                )
        }
        row {
            text(
                "<b>Why GitHub CLI is needed:</b><br/>" +
                    "The Copilot ACP (Agent Communication Protocol) mode does not expose billing " +
                    "or usage data. To view Copilot premium request usage, the plugin uses the " +
                    "GitHub CLI (<code>gh</code>) to query GitHub's internal API endpoint.",
                MAX_LINE_LENGTH_WORD_WRAP
            ).applyToComponent { foreground = UIUtil.getContextHelpForeground() }
        }
        row {
            text(
                "Install from <a href='https://cli.github.com/'>cli.github.com</a>, " +
                    "then authenticate with <code>gh auth login</code>.",
                MAX_LINE_LENGTH_WORD_WRAP
            ).applyToComponent { foreground = UIUtil.getContextHelpForeground() }
        }
        row {
            val link = HyperlinkLabel("Install GitHub CLI")
            link.setHyperlinkTarget("https://cli.github.com/")
            cell(link)
        }
        separator()
        row {
            checkBox("Show Copilot usage graph in toolbar")
                .comment(
                    "Shows a usage graph icon in the main toolbar when Copilot usage data is available."
                )
                .bindSelected(
                    { settings.isShowCopilotUsage },
                    { settings.isShowCopilotUsage = it }
                )
        }
    }

    override fun reset() {
        super<BoundConfigurable>.reset()
        refreshGhCliStatusAsync()
    }

    private fun refreshGhCliStatusAsync() {
        statusLabel.text = "Checking..."
        statusLabel.foreground = UIUtil.getLabelForeground()
        ApplicationManager.getApplication().executeOnPooledThread {
            val client = CopilotBillingClient()
            val ghCli = client.findGhCli()
            val authenticated = ghCli != null && client.isGhAuthenticated(ghCli)
            ApplicationManager.getApplication().invokeLater {
                when {
                    ghCli == null -> {
                        statusLabel.text = "GitHub CLI not found — install from cli.github.com"
                        statusLabel.foreground = JBColor.RED
                    }
                    !authenticated -> {
                        statusLabel.text =
                            "GitHub CLI found but not authenticated — run 'gh auth login'"
                        statusLabel.foreground = JBColor(0xFF8C00, 0xFFA040)
                    }
                    else -> {
                        statusLabel.text = "✓ GitHub CLI authenticated"
                        statusLabel.foreground = JBColor(0x008000, 0x4EC94E)
                    }
                }
            }
        }
    }

    companion object {
        const val ID = "com.github.catatafishen.agentbridge.billing"
    }
}
